dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java

```diff
@@ -1494,12 +1499,32 @@ public final Response fireActionDefault(@Context final HttpServletRequest reques
         }
     } // fireAction.
 
+    private void checkContentletState(final Contentlet contentlet, final SystemAction systemAction)
+            throws NotFoundInDbException {
+
+        if (null == contentlet) {
+
+            throw new NotFoundInDbException("Not Contentlet Found");
+        }
+
+        if (contentlet.isNew()) {
+
+            if (    systemAction == SystemAction.UNPUBLISH ||
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4324168)